### PR TITLE
MAINT: Remove yield tests

### PR DESCRIPTION
**Pull request by charris**
Removes the remaining yield tests deprecated by pytest. The travis and appveyor tests are also adjusted to no longer ignore the pytest warnings.

The polynomial classes tests were parametrized, in the remaining cases the yield statements were simply replaced by function calls. The latter seems a bit crude, but the tests were not designed to be parametrized and there were also useful comments in the code that would have been lost in doing so. Because each yielded test was counted as separate, whereas the function calls are not, there has been a net decrease in the reported number of tests run.

There are also some style fixups, could not help myself.

### CODE REVIEW

File: numpy/core/tests/test_umath_complex.py [hunk 1]
```diff
@@ -107,18 +107,18 @@ def _check_inf_nan(dummy):
                 if not np.isinf(z.real) or not np.isnan(z.imag):
```
Comment (mhvk) [inline numpy/core/tests/test_umath_complex.py]: Really beyond the PR, but here and below, I think the following is clearer (no "double negatives")
```
assert np.isinf(z.real) and np.isnan(z.imag), msgform % (z.real, z.imag)
```

File: numpy/polynomial/tests/test_classes.py [hunk 2]
```diff
@@ -19,41 +20,48 @@
...
+    )
+classids = (
+    'Polynomial', 'Legendre', 'Chebyshev', 'Laguerre',
+    'Hermite', 'HermiteE'
+    )
+
+
+
+
+#def test_class_methods():
```
Comment (mhvk) [inline numpy/polynomial/tests/test_classes.py]: To be done? Shouldn't have commented out code left in files...

File: numpy/polynomial/tests/test_classes.py [hunk 3]
```diff
@@ -116,7 +127,8 @@ def check_cast(Poly1, Poly2):
 #
 
 
-def check_identity(Poly):
+@pytest.mark.parametrize('Poly', classes, ids=classids)
```
Comment (mhvk) [inline numpy/polynomial/tests/test_classes.py]: I think the "pytest way" to do this repeatedly is to define a "fixture" -- must admit I don't always find it as clear, but it is definitely less repetitive. Something like:
```
@pytest.fixture(params=classes, ids=classids)
def Poly(request):
    return request.param

def test_identity(Poly):
    ...
```

See https://docs.pytest.org/en/latest/fixture.html#fixture-parametrize

**Review by mhvk — COMMENTED** [review]
This looks good, thanks for the thankless work...

Comments are just suggestions; the last one perhaps the one that would a bit of a difference.

File: numpy/lib/tests/test_index_tricks.py [hunk 4]
```diff
@@ -284,84 +284,90 @@ def test_c_():
...
-           np.array([[[1, 0],
-                   [0, 0]],
-
-                  [[0, 0],
-                   [0, 1]]]))
+    assert_array_equal(
+        a, np.array([[[1, 0],
+                      [0, 0]],
+                    [[0, 0],
+                     [0, 1]]])
```
Comment (eric-wieser) [inline numpy/lib/tests/test_index_tricks.py]: nit: Alignment is off a little here

File: numpy/polynomial/tests/test_classes.py [hunk 5]
```diff
@@ -19,41 +20,48 @@
...
-        yield check_cutdeg, Poly
-        yield check_truncate, Poly
-        yield check_trim, Poly
-        yield check_ufunc_override, Poly
+    Hermite, HermiteE
+    )
+classids = (
+    'Polynomial', 'Legendre', 'Chebyshev', 'Laguerre',
+    'Hermite', 'HermiteE'
+    )
```
Comment (eric-wieser) [inline numpy/polynomial/tests/test_classes.py]: Better as `tuple(cls.__name__ for cls in classes)`

File: numpy/polynomial/tests/test_classes.py (hunk 2 above)
Comment (eric-wieser) [reply numpy/polynomial/tests/test_classes.py]: Looks like this is done, and just needs removing.

File: numpy/polynomial/tests/test_classes.py (hunk 3 above)
Comment (charris) [reply numpy/polynomial/tests/test_classes.py]: Ah, I see what pytest has done: "One fixture to rule them all ..."

File: numpy/core/tests/test_umath_complex.py (hunk 1 above)
Comment (charris) [reply numpy/core/tests/test_umath_complex.py]: Yep, the `test_umath_complex.py` module could use a major refactoring. I sort of started down that road before stopping myself :)

Comment (charris) [reply numpy/core/tests/test_umath_complex.py]: Note that the functions were created so that they could be used with the previous `yield` statement.

**Comment by charris** [comment]
Fixed.

**Pull request merged**